Doc/library/collections.rst

```diff
@@ -1252,16 +1252,17 @@ variants of :func:`functools.lru_cache`:
                 self.cache.move_to_end(args)
                 return self.cache[args]
             result = self.func(*args)
-            self.requests[args] = self.requests.get(args, 0) + 1
+            self.requests[args] = self.requests.get(args, 0) + 1 # increment existing value by 1 if seen before, otherwise set to 1
             if self.requests[args] <= self.cache_after:
                 self.requests.move_to_end(args)
                 if len(self.requests) > self.maxrequests:
-                    self.requests.popitem(0)
+                    self.requests.popitem(last=False)
             else:
-                self.requests.pop(args, None)
+                # entry has been seen more than cache_after times
+                self.requests.pop(args) # no longer need to keep track of how many times this entry has been seen
```

Comment:
This comment is reasonable and add some value.  Please move it to a separate line, just before the relevant code.

Reply:
I moved this comment to the lines just above the code in commit 77f10b2a66b2d7353ccb13783021b459484e0de4.

I also added context the multi-threading problems that could arise that you mentioned in commit a03483d5cd9cc3956f4f4bae51c416dc67d1fb96, since I think the race conditions that could occur in multi-threading environments is kind of obvious for someone who is already designing code for multi-threading environments - but might be missed by someone who _**initially designs the code for a single-threaded environment**_ and then moves it into a multi-threaded environment without reviewing all the code.

I think this is useful context and might help some people avoid problems if they mistakenly take out the default argument from `pop(args, None)` because they think it is unnecessary (in their single-threaded environment), but let me know if you think the comment is too long/should be changed, or if the comment is unnecessary altogether.

Reply:
Please remove the comments altogether.   Just stick with the 1/0 to True/False edit.

Reply:
I think this is related to https://github.com/python/cpython/pull/107613#discussion_r1284981437, in which case, commit aae218b39b27f09a39ed0cd6fba09cee26aafbea should take care of this.